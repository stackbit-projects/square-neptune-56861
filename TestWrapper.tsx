import React from 'react';
import { ThemeProvider } from "styled-components";
import { theme } from './theme';

export const TestWrapper = (props) => {
 return (
  <ThemeProvider theme={theme}>
    {props.children}
  </ThemeProvider>
 )
}
