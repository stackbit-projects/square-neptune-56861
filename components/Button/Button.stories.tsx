import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';

import Button from '.';

export default {
  title: 'Button | CTAButton',
  component: Button,
} as Meta;

const Template: Story = (args) => <Button  {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  renderingContext: {
    id: '1234',
    item: {
      fields: {
        link: {
          url: "/",
          text: "click me"
        }
      }
    }
  }
};
