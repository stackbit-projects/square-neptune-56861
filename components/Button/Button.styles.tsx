import styled from 'styled-components'
import { typography } from '../../theme'

export const ContinueButton = styled.button`
  ${typography.link}
  background: ${props => props.theme.colors.primaryLight};
  border-radius: 25px;
  padding: 13px 50px 13px 25px;
  background-size: 16px 16px;
  background-repeat: no-repeat;
  background-position: center right 24px;
  background-image: url(/images/icon-arrow-blue.svg);
`