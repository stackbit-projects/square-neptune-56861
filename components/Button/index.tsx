import React from 'react'
import { linkFormatter } from '../../utils/formatter'

const Variants = {
  Plain: "{1231EFBC-DE7A-489F-9410-C25C2B2DEBBB}",
  Large: "{69244147-9040-432C-A6AC-16BAE4863271}",
  Medium: "{859E750E-BF54-4DA2-8DD4-9786565E3640}",
  Small: "{19AE9B21-88B6-4D17-82B1-CA605DB70EA7}",
  Link: "{0D07C7F0-4367-4647-BFAF-8AAD9A8B1E27}",
}

const Button = (props) => {
  const { renderingContext } = props

  if (!renderingContext.item || !renderingContext.item.fields || !renderingContext.item.fields.link) {
    console.error("No item supplied for Button")
    return null
  }

  const { item: { id, fields: { link } }, parameters } = renderingContext

  const isPlain = parameters?.FieldNames === Variants.Plain;
  const isLarge = parameters?.FieldNames === Variants.Large;
  const isMedium = parameters?.FieldNames === Variants.Medium;
  const isSmall = parameters?.FieldNames === Variants.Small;
  const isLink = parameters?.FieldNames === Variants.Link;

  let buttonClass = isPlain ? "cta-plain" : ""

  if (isLarge) {
    buttonClass = "cta-large"
  }

  if (isMedium) {
    buttonClass = "cta-medium"
  }

  if (isSmall) {
    buttonClass = "cta-small"
  }

  if (isLink) {
    buttonClass = "cta-link"
  }

  const url = linkFormatter(link)

  return (
    <div className="component c-ctaButton ctaWrapper columns c-header__cta">
      <div className="component-content">
        <div className="field-link">
          <a href={url} data-variantitemid={`{${id}}`} className={buttonClass} role="button" data-variantfieldname="Link" target={link.target}>{link.text}</a>
        </div>
      </div>
    </div>
  )
}

export default Button