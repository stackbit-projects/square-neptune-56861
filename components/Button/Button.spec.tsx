import React from 'react';
import renderer from 'react-test-renderer';
import Button from '.';
import { TestWrapper } from '../../TestWrapper';

it('Button renders correctly', () => {
  const tree = renderer
    .create(<TestWrapper>
      <Button renderingContext={{
        page: {
          renderings: [{
            settings: {
              DataSource: ''
            }
          }]
        },
        item: {
          id: 'test',
          fields: {
            link: {
              url: "/",
              text: "click me"
            }
          }
        }
      }} />
    </TestWrapper>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});