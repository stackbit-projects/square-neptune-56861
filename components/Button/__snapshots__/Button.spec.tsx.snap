// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<div
  className="component c-ctaButton ctaWrapper columns c-header__cta"
>
  <div
    className="component-content"
  >
    <div
      className="field-link"
    >
      <a
        className=""
        data-variantfieldname="Link"
        data-variantitemid="{test}"
        href="/"
        role="button"
      >
        click me
      </a>
    </div>
  </div>
</div>
`;
